chia/_tests/util/test_service_groups.py

```diff
@@ -0,0 +1,16 @@
+from __future__ import annotations
+
+from chia.util.service_groups import services_for_groups
+
+
+def test_services_for_groups() -> None:
+    i = 0
+    for service in services_for_groups(["harvester"]):
+        assert service == "chia_harvester"
+        i += 1
+    assert i == 1
+
+    i = 0
+    for _ in services_for_groups(["daemon"]):
+        i += 1
+    assert i == 0
```

Comment:
Correct. What I wanted to check is to confirm that the loop in the below picture never occurs for `chia start daemon`
![image](https://github.com/Chia-Network/chia-blockchain/assets/84098616/a5f8cca0-ed4d-4c87-90d3-83382ff9ebf9)

Your code suggestion looks clean but I think it's not completely the same as my code since `list(...)` forcibly converts generator into a list but I wanted to check that `for ...` against the empty generator doesn't execute the content in the `for ...` loop.
But I don't like to be a stubborn about this.
Push me one more time and I'll commit your suggestion.

Reply:
Thanks for the explanation, I can see how matching the form of the runtime code is of interest.  A for loop not executing the body for an 'empty' iterable, generator or otherwise, is fairly core to its functionality so personally I don't feel the need to check that.  But I understand your interest.  Do as you will here and feel free to resolve this conversation after having read this comment.